src/components/form/form.types.ts

```diff
@@ -33,3 +35,72 @@ export interface FormError {
      */
     message: string;
 }
+
+export interface FormComponent<T = any> {
+    /**
+     * The value of the current property
+     */
+    value: T;
+
+    /**
+     * Whether or not the current property is required
+     */
+    required?: boolean;
+
+    /**
+     * Whether or not the current property is readonly
+     */
+    readonly?: boolean;
+
+    /**
+     * Whether or not the current property is disabled
+     */
+    disabled?: boolean;
+
+    /**
+     * The name of the current property
+     */
+    name?: string;
```

Comment:
Maybe this belongs to the `FormInfo` interface? :thinking: 
All of the other properties here are used to change the appearance/behavior of the field in some way, but this seems to be more of a "metadata" type field. What do you think?

Reply:
ya makes sense